src/native_ext/profiling.cpp

```diff
@@ -247,29 +247,21 @@ ActivationBin* ProfilingGetActivationBin(Profiling* profiling, int64_t binIndex)
 }
 
 SpanActivation* FindClosestActivation(Profiling* profiling, int64_t ts) {
-  SpanActivation sentinel;
-  sentinel.startTime = std::numeric_limits<int64_t>::min();
-  sentinel.endTime = std::numeric_limits<int64_t>::max();
-  SpanActivation* t = &sentinel;
-
+  SpanActivation* t = nullptr;
```

Comment:
What was your g++ version? The compiler is wrong to produce a warning here as there was a check for the sentinel value before returning. Does it happen in CI as well? This loop is the hottest path when doing span matching

Reply:
`g++ (GCC) 12.1.1 20220507`.
Yeah, I see that it's checked for, but assumed the compiler didn't catch that.